### DISCUSSION TITLE
How to list included media files

### DISCUSSION THREAD

**Discussion by simeonovp** [thumbsup/thumbsup#332] [Q&A] [answered]
I wont to generate album containing selected media files. For examle the folder "/home/user/Album/2023" contains 5 images IMG_x.JPG, where x is 1 to 5. I wont skip 2 and 4. My config lok like:
{
  "input": "/home/user/Album",
  "output": "/home/user/Web",
  "include": [
    "2022/IMG_0.JPG",
    "2023/IMG_1.JPG",
    "2023/IMG_3.JPG",
    "2023/IMG_5.JPG"
  ]
}
On generate no files are detected: **1 albums, 0 photos, 0 videos**
This is a simplified example. In fact, I have several hundred directories with photos and generate the include lists with tool.
What is the solution for this?

**Comment by rprieto** [comment]
Hi, this is strange because your config looks OK.
Can you show an extract of the folder listing (`ls -la`)?
Is the case exactly the same (e.g. `JPG` vs `jpg`)?

**Reply by simeonovp** [reply]
Hi rprieto, thanks for your anwer. The problem is not with the file names but with the paths. If I define the files as follow then this likes fine, but this way I can't avoid naming conflicts in subfolders. If I replace both stars with concrete paths it doesn't work again. Also 2023*/... not match
```
{
"input": "/home/user/Album",
"output": "/home/user/Web",
"include": [
"**/IMG_0.JPG",
"**/IMG_1.JPG",
"**/IMG_3.JPG",
"**/IMG_5.JPG"
]
}
```

**Reply by simeonovp** [reply]
Hi rprieto, I think I found the problem. This configuration cannot work because all items from "include" are relative to "input" (which would be more intuitive for me) and the search (glob) interprets them as relative to CWD.

**Reply by rprieto** [reply]
Thank you, that's an interesting discovery. Can you please edit your previous messages with the `code` syntax (three backticks) so we can understand your glob patterns better? It could be a bug in thumbsup, and in this case we should be able to write a test around it.

**Reply by simeonovp** [reply]
I must correct myself. The configuration for this example that works looks like this:
```
{
"input": "/home/user/Album",
"output": "/home/user/Web",
"include": [
"2022/",
"2022/IMG_0.JPG",
"2023/",
"2023/IMG_1.JPG",
"2023/IMG_3.JPG",
"2023/IMG_5.JPG"
]
}
```
If the directories are not explicitly listed, they will not be scanned recursively

**Reply by rprieto** [reply]
Thanks. So to make sure I understand, neither of these work on their own?
* `**/IMG_0.JPG`
* `2022/IMG_0.JPG`

I will run some tests too.

**Comment by simeonovp** [comment]
If the media file includes starts with two asterisks, it works, but it is not unique, since two files with the same name can exist in different subdirectories. A definition with (one or more) subfolders (subdir1/file.ext, subdir1/subdir2/file.ext) only works if include entries for the subfolders also exist (subdir1/, subdir1/subdir2/). The reason is that the stream from "readdir-enhanced" uses the same options (including filter method) for directories and files. If a directory does not find a match, it is not searched deeper.
The solution is OK. But it would be nice if an example is offered for usage.

**Reply by rprieto** [reply, marked as answer]
Hi @simeonovp. Thanks for all these details, that was exactly the problem. The good news is that it's now fixed on the main branch. I did some testing and it's looking OK now. It will be part of the next release.
